haskell/toolchain.bzl

```diff
@@ -287,7 +287,13 @@ _haskell_toolchain = rule(
       doc = """
 Label pointing to the locale archive file to use. Mostly useful on NixOS.
 """,
-    )
+    ),
+    "_dummy_static_lib": attr.label(
```

Comment:
I think you can't put this here because to generate a dummy archive you need a toolchain and a toolchain needs a dummy archive now. This creates a dependency cycle which leads to the failure you can see on CI.

Reply:
Yes, I also got this:

```
$ bazel query 'deps(//tests:test-binary-with-lib)'
//tests:test-binary-with-lib
@bazel_tools//tools/test:test_setup
@bazel_tools//tools/test:runtime
@bazel_tools//tools/test:test-setup.sh
@bazel_tools//tools/test:collect_coverage
//tools/defaults:coverage_support
@bazel_tools//tools/test:coverage_support
@bazel_tools//tools/test:collect_coverage.sh
//tools/defaults:coverage_report_generator
@bazel_tools//tools/test:coverage_report_generator
@bazel_tools//tools/coverage:coverage_report_generator
@bazel_tools//tools/coverage:dummy_coverage_report_generator
//tests/binary-with-lib:binary-with-lib
//tests/binary-with-lib:lib
@io_tweag_rules_haskell//haskell:private/ghci_repl_wrapper.sh
@io_tweag_rules_haskell//haskell:assets/ghci_script
@io_tweag_rules_haskell//haskell:assets/cpp_defines
//tests/binary-with-lib:src/Lib.hs
//tests/binary-with-lib:Main.hs
```

Which doesn’t include the target as I would expect.

Reply:
So to make it work you'll need to remove this from the toolchain and perhaps reference that dummy archive from `compile_binary` like so: `@io_tweag_rules_haskell//haskell:dummy_static_lib`. That may work.

Reply:
Ah, I think it have to add it to `_mk_binary_rule` instead of the toolchain.